### ISSUE TITLE
ChatGPT for business [List of usecases]

### ISSUE THREAD

**Issue by two-trick-pony-NL** [two-trick-pony-NL/DocBotGPT#1]
# TL:DR
Rough idea is as follows: 
- A Finn copy 
- So a chat window trained with company specific data. I created this issue to track all he usecases I thought of that you could feed such a system

# What sources
- Finance data (So you feed invoices sent and received) so you can query when did we last order from shop X
- Email data
- Calendar data
- PDF document with business related logic
- Customer support data 
- Social media posts 
- Marketing email lists
- Spreadsheet / word docs
- Chat/Slack data of public channels
- code
- contracts
- employee records
- candidates recruitment
- Customer reviews
- Blog (by URL)
- website (by URL)
- meeting transcripts
- analytics databases
- Laws
- On daycare: The instructions for kids on daycare. --> So parents write the preference for the kids

^ All of these sources should be normalised and metadata collected (how many emails were sent, how many candidates etc) 

# Proof of concept code
- https://github.com/techleadhd/chatgpt-retrieval
- https://www.youtube.com/watch?v=9AXP7tCI9PI

Library:
- https://python.langchain.com/docs/get_started/introduction

# Potential customers: 
### Charactaristics
- Any company that does something according to a specific method
- companies that have their own policies
- Customer support first line
- Sits behind computer

### Names
- Loo van Eck - helps big companies write better letters
- Municipalities / Government --> Gemeente with their local regulations either inform customers better or for internal employees (If we aim at citizens it might mean a pivot) 
- 

# Further ideas: 
- Perhaps a 'company wide'  and a 
- Individual employee one 
- Integrated with chat apps
- Should have a log of which user prompts
- have it as a desktop app like slack. So you have your personalised chat and documents but also get a feed of the most important stuff
- This way you can set up recurring queries. Send me a summary of X every monday at 0900. Give me the revenue at the end of the quarter. 

# Quick sketch

![Image](https://github.com/two-trick-pony-NL/DocBotGPT/assets/71013416/48530664-4347-439b-8882-25ff13267968)

**Comment by two-trick-pony-NL** [comment]
Made a MVP here: https://docbotgpt.streamlit.app/ 
Pretty cool.
